src/rdkafka_msg.c

```diff
@@ -169,14 +169,13 @@ static rd_kafka_msg_t *rd_kafka_msg_new0 (rd_kafka_itopic_t *rkt,
         if (hdrs)
                 hdrs_size = rd_kafka_headers_serialized_size(hdrs);
 
-	if (unlikely(len + keylen + hdrs_size >
-		     (size_t)rkt->rkt_rk->rk_conf.max_msg_size ||
-		     keylen > INT32_MAX)) {
-		*errp = RD_KAFKA_RESP_ERR_MSG_SIZE_TOO_LARGE;
-		if (errnop)
-			*errnop = EMSGSIZE;
-		return NULL;
-	}
+        if (unlikely(rd_kafka_msg_max_wire_size(keylen, len, hdrs_size) >
```

Comment:
Should keylen still be restricted to 32bits?

Reply:
Yes man! And `len`, to avoid overflow in max_wire..